index.d.ts

```diff
@@ -1,11 +1,11 @@
-export type TimeComponents = {
-	days: number;
-	hours: number;
-	minutes: number;
-	seconds: number;
-	milliseconds: number;
-	microseconds: number;
-	nanoseconds: number;
+export type TimeComponents<T extends number | bigint> = {
```

Comment:
Default it to `number` to not break backwards compatibility and also optimize for the common case.
```suggestion
export type TimeComponents<T extends (number | bigint) = number> = {
```